nntrainer/compiler/ini_interpreter.cpp

```diff
@@ -39,10 +39,88 @@ namespace {
 class PlainLayer {};    /**< Plain Layer tag */
 class BackboneLayer {}; /**< Backbone Layer tag */
 
-template <typename T>
-static std::shared_ptr<Layer> section2layer(dictionary *ini,
+/**
+ * @brief convert section to list of string based properties
+ *
+ * @param ini ini handler
+ * @param sec_name section name
+ * @return std::vector<std::string> list of properties
+ */
+std::vector<std::string> section2properties(dictionary *ini,
                                             const std::string &sec_name) {
-  /// NYI!
+  int num_entries = iniparser_getsecnkeys(ini, sec_name.c_str());
+  NNTR_THROW_IF(num_entries < 1, std::invalid_argument)
+    << "there are no entries in the layer section: " << sec_name;
+  ml_logd("number of entries for %s: %d", sec_name.c_str(), num_entries);
+
+  std::unique_ptr<const char *[]> key_refs(new const char *[num_entries]);
+  NNTR_THROW_IF(iniparser_getseckeys(ini, sec_name.c_str(), key_refs.get()) ==
+                  nullptr,
+                std::invalid_argument)
+    << "failed to fetch keys for the section: " << sec_name;
+
+  std::vector<std::string> properties;
+  properties.reserve(num_entries - 1);
+
+  for (int i = 0; i < num_entries; ++i) {
+    std::string key(key_refs[i]);
+    std::string prop_key = key.substr(key.find(":") + 1);
+
+    if (istrequal(prop_key, "type") || istrequal(prop_key, "backbone")) {
```

Comment:
please follow the latest `model loader` code style where the list of properties to be skipped is given by each section - layer, backbone, optimizer, etc have different properties to be skipped.

Reply:
I checked this restriction and it works or fine with model_loader. please get me right if I am missing something.

Reply:
My earlier concern was that `type` and `backbone` can be passed externally as parameters, but as this is being used only for parsing layers, its fine.